core/state_processor.go

```diff
@@ -309,7 +309,9 @@ func (p *StateProcessor) Process(block *types.WorkObject, etxSet *types.EtxSet)
 					return nil, nil, nil, nil, 0, fmt.Errorf("etx %032x emits UTXO with value %d greater than max denomination", tx.Hash(), tx.Value().Int64())
 				}
 				// There are no more checks to be made as the ETX is worked so add it to the set
-				statedb.CreateUTXO(tx.OriginatingTxHash(), tx.ETXIndex(), types.NewUtxoEntry(types.NewTxOut(uint8(tx.Value().Int64()), tx.To().Bytes())))
+				if err := statedb.CreateUTXO(tx.OriginatingTxHash(), tx.ETXIndex(), types.NewUtxoEntry(types.NewTxOut(uint8(tx.Value().Int64()), tx.To().Bytes()))); err != nil {
```

Comment:
Going from big.Int to Int64 to uint8 seems sus. Can you confirm our values never exceed uint8?

Reply:
The check for this is above: 
```if tx.Value().Int64() > types.MaxDenomination { // sanity check
					return nil, nil, nil, nil, 0, fmt.Errorf("etx %032x emits UTXO with value %d greater than max denomination", tx.Hash(), tx.Value().Int64())```

Reply:
Ok since we do the overflow sanity check is there anywhere we check that the value is strictly above 0?

Reply:
Making it a uint64 would fix it too. That would simultaneously check that the value is below the max denomination and that it was positive.

Reply:
https://github.com/dominant-strategies/go-quai/compare/e3e572656a9b5f5944d142ec4c1ec18e09c704f1..7f802f01cd2b5342c48f850e7551c99fbaddf492

Reply:
Updated the conversion PR to use big.Int.Uint64()